core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -126,3 +126,8 @@ trait RequireAsyncReset extends Module {
 trait RequireSyncReset extends Module {
   override private[chisel3] def mkReset: Bool = Bool()
 }
+
+trait HasParameterizedResetType extends Module {
+  val hasAsyncNotSyncReset: Boolean
```

Comment:
```suggestion
trait HasParameterizedResetType extends Module {
  def hasAsyncNotSyncReset: Boolean
```
virtual methods should be defs

Also, I don't think "HasParameterizedResetType" is the right name for this. HasParameterizedResetType implies to me an abstract reset rather than saying "you must now pick with this val. I think this is a good thing to add, I just would like a different name.

Another option is to just make `mkReset` public (or protected) and then users can pick the reset type directly.